### DISCUSSION TITLE
Note Event on Channel

### DISCUSSION THREAD

**Discussion by suce-ex** [RomanKubiak/ctrlr#458] [Q&A]
Hi community, 

simple question:

what is lua code for "note event (on channel) "
to get
if (any) note event (on ch1, or ALL ch) do

?
Thanks , and where I could have found that info elsewhere

**Comment by DRBAUS** [comment]
Hi suce-ex,

Do you really need a lua code or will the MIDI Message type in the Panel Mode area set to MIDI Note On suffice?

**Reply by suce-ex** [reply]
Hi @DRBAUS , good question, i didnt know about this section. But I would need the lua code :=)

**Reply by suce-ex** [reply]
I wish to do in LUA: 
- anything received on NOTE EVENT (CHANNEL: 1 or 2 or ALL) [lua code] > does something > [my lua code]

**Reply by damiensellier** [reply]
Note on/off is 0x80 to 0x9F on the 2nd byte of the midi message.
You need a global script on "Called when the panel receives a midi  message"
<img width="465" alt="Capture d’écran 2022-04-25 à 06 47 47" src="https://user-images.githubusercontent.com/55164385/165074800-cbb1dfaf-9aa1-4616-bee8-37ff89f23e1f.png">

```
--
-- Called when a panel receives a midi message (does not need to match any modulator mask)
-- @midi   CtrlrMidiMessage object
--

midiMessageReceived = function(MidiMessage)

	local midiMessageType = MidiMessage:getData():getByte(2) -- get the 2nd byte of the received message

	if	midiMessageType  >= 0x80 and midiMessageType <= 0x9F then -- Note ON/OFF is 0x80 to 0x9F where the low nibble is the MIDI channel. 

		-- do your thing here

	end


end
```



If you need only on note ON or Off you have to narrow down the statement to  0x80 to 0x8F (for NOF) or 0x90 to 0x9F (for NON)
<img width="808" alt="Capture d’écran 2022-04-25 à 06 58 26" src="https://user-images.githubusercontent.com/55164385/165076176-e4c32387-6a44-481b-afbd-0125eafb5051.png">

**Reply by suce-ex** [reply]
@damiensellier , @DRBAUS , 

I got it! works, (9F is Note ON Ch16) thank you both!
@damiensellier ,
you probably meant getByte(0) to get our 1st byte (just for other spectators).

**Reply by damiensellier** [reply]
@suce-ex You're right!

**Reply by suce-ex** [reply]
And you are ALL excellent !!! <3

**Reply by suce-ex** [reply]
Extension Question: 

C2 is : 90 30 xx

how to command it calls the 5th on CHannel2 which would be 
G2 is : 91 37 xx
(+7 for fifth)
?

---
[with what kind of command : add byte (1) "+7 
and
add byte (0) +1 
???]

**Reply by suce-ex** [reply]
update:

what i was trying to ask is as KSP (NI Kontakt Script)
somehow this idea:

```
on midi_in
if ($MIDI_COMMAND = $MIDI_COMMAND_NOTE_ON and $MIDI_BYTE_2 > 0)
set_midi
($MIDI_CHANNEL,$MIDI_COMMAND_NOTE_ON,$MIDI_BYTE_1+4,$MIDI_BYTE_2)
set_midi
($MIDI_CHANNEL,$MIDI_COMMAND_NOTE_ON,$MIDI_BYTE_1+7,$MIDI_BYTE_2)
end if
if ($MIDI_COMMAND = $MIDI_COMMAND_NOTE_OFF or ...
($MIDI_COMMAND = $MIDI_COMMAND_NOTE_ON and $MIDI_BYTE_2 = 0))
set_midi ($MIDI_CHANNEL,$MIDI_COMMAND_NOTE_ON,$MIDI_BYTE_1+4,0)
set_midi ($MIDI_CHANNEL,$MIDI_COMMAND_NOTE_ON,$MIDI_BYTE_1+7,0)
end if
end on
```

this makes it a simple harmonizer.
